elm/src/test/scala/io/scalac/elm/functional/FuzzyTransactionTest.scala

```diff
@@ -0,0 +1,141 @@
+    Http(uri).header("Accept", "text/plain").option(HttpOptions.readTimeout(1000)).asString.body.toInt
+  } getOrElse (throw new IllegalStateException("Funds should be always accessible"))
+
+  private def makePayment(sender: TestElmApp, address: String, amount: Int, priority: Int) = Try {
+    val uri = s"http://localhost:${node2port(sender)}/wallet/payment"
+    sender.log.debug(s"Trying to query wallet address: $uri")
+    Http(uri).header("Accept", "text/plain").option(HttpOptions.readTimeout(1000))
+        .param("address", address)
+        .param("amount", amount.toString)
+        .param("priority", priority.toString)
+        .asString.body
+  }
+
+  override protected def beforeAll(): Unit = nodes foreach { node =>
+    @tailrec
+    def waitForInitialization: Unit = {
+      node.synchronized(node.wait(100))
+      getWalletAddress(node) match {
+        case Success(address) if address.nonEmpty || address == "0" =>
+        case Success(address) =>
+          node.log.debug(s"Returned empty message")
+          waitForInitialization
+        case Failure(error)   =>
+          node.log.debug(s"Returned error message $error")
+          waitForInitialization
+      }
+    }
+
+    node.run()
+    waitForInitialization
+    node.log.info(s"Node initialized for testing")
+  }
+
+  "Series of random transactions requests" should "result in successfully finalized transactions" in {
+
+    val addresses = nodes map getWalletAddress map (_.getOrElse(
+      throw new IllegalStateException("All addresses should be known at this point")))
+    val node2Address = nodes zip addresses toMap
+
+    nodes foreach { node =>
+      node.log.trace(s"My address is ${node2Address(node)}")
+    }
+
+    @tailrec
+    def performRandomTransactions(node2State: Map[TestElmApp, NodeExpectedState], transactionsLeft: Int):
+        Map[TestElmApp, NodeExpectedState] = {
+
+      if (transactionsLeft <= 0) node2State
+      else {
+        val maxFunds = node2State.values map (_.funds) max
+        val priority = Random.nextInt(maxFunds / 10 / 10)
+        val fee = priority * 10
+        val maxAmount = ((maxFunds - fee) * 0.3).toInt
+        val amount = Random.nextInt(maxAmount)
+
+        val sender = Random.shuffle(node2State filter { case (_, s) => s.funds >= amount + fee }).head._1
+        val receiver = Random.shuffle(nodes).head
+        val receiverAddress = node2Address(receiver)
+
+        makePayment(sender, receiverAddress, amount ,fee)
+        sender.log.debug(s"Sent $amount with $fee of fee to $receiverAddress")
+
+        // TODO: figure out who created block and received fee
+
+        val newStates = node2State
+          .fundsChangedBy(sender, -(amount + fee))
+          .fundsChangedBy(receiver, amount)
+
+        synchronized(wait(100))
+
+        performRandomTransactions(newStates, transactionsLeft -1)
+      }
+    }
+
+    val initialNode2State = nodes zip (nodes map getWalletFunds map NodeExpectedState) toMap
```

Comment:
I think we need another wait here to allow nodes to sync chain and than check state of wallets